test/include/rev-macros.h

```diff
@@ -13,7 +13,7 @@
 #define TRACE_POP      asm volatile("slli x0,x0,4");
 #define TRACE_ASSERT(x) { TRACE_PUSH_ON; \
   if (!(x)) { asm volatile(".word 0x0"); }; \
-  TRACE_PUSH_OFF }
+  TRACE_POP }
```

Comment:
I would put `do{ }while(0)` around the macro definition so that it works like a function call and can be properly nested inside `if` statements, etc.